orttraining/orttraining/models/runner/training_util.cc

```diff
@@ -72,6 +72,29 @@ size_t DataSet::TotalBatch(size_t batch_size) const {
   return NumSamples() / batch_size + ((NumSamples() % batch_size > 0) ? 1 : 0);
 }
 
+// gather additional training params from tensor dimensions
+// see metrics_map in bert/main.cc for example, and training_utils.h for more explanation
+common::Status DataSet::GetTensorDimensionsFromInputs(const std::map<std::string, std::pair<std::string, size_t>>& metrics_map,
+                                                      MapStringToString& perf_properties) const {
+  if (metrics_map.size() == 0) return Status::OK(); 
+
+  for (size_t input_index = 0; input_index < NumInputs(); ++input_index) {
+    std::string input_name = GetInputName(input_index);
+    const auto it = metrics_map.find(input_name);
+    if (it == metrics_map.end()) continue;
+    auto metric = it->second;
+
+    const Tensor& first_tensor = data_[0]->at(input_index).Get<Tensor>();
+    std::vector<int64_t> shape_vector = first_tensor.Shape().GetDims();
+
+    if (metric.second < shape_vector.size())
+      perf_properties.insert({metric.first, std::to_string(shape_vector[metric.second])});
+    else 
+      return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "shape position out of bounds for input: ", input_name.c_str());
```

Comment:
could also put the requested index and the actual size. consider using ORT_RETURN_IF_NOT()